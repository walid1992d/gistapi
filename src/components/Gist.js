import styled from "styled-components";
import Octicon from 'react-octicon'
const formatDate = (rawDate) => {
    const date = new Date(rawDate);

    return `${date.getMonth()+1}/${date.getDay()}/${date.getFullYear()}`
}
const Gist = ({ gist }) => <GistWrapper>
    <Header>
        <Creator>
        <IconWrapper  href={gist.owner.html_url} target="__blank">
        <Avatar src={gist.owner.avatar_url}></Avatar>
        <Name>{gist.owner.login}</Name>
        </IconWrapper>
   
        </Creator>
        <IconsWrapper>

            <IconWrapper>
            <Octicon name="code"></Octicon>
                <IconTitle> {Object.keys(gist.files).length} Files</IconTitle>
            </IconWrapper>
            <IconWrapper href={gist.forks_url} target="__blank">
            <Octicon name="repo-forked"></Octicon>
                <IconTitle> Forks</IconTitle>
            </IconWrapper>
            <IconWrapper  href={gist.comments_url} target="__blank">
            <Octicon name="comment"></Octicon>
                <IconTitle> Comments</IconTitle>
            </IconWrapper>
            <IconWrapper href={gist.owner.starred_url} target="__blank">
            <Octicon name="star" ></Octicon>
                <IconTitle> Stars</IconTitle>
            </IconWrapper>
        </IconsWrapper>

        
        
    </Header>
    <Dates>
        Created at: {formatDate(gist.created_at)} &nbsp; &nbsp; Last updated: {formatDate(gist.updated_at)}
    </Dates>
    <Description>
        {gist.description}
    </Description>

    <FilesWrapper>
        {Object.keys(gist.files).map((fileName) => {
            const currentFile = gist.files[fileName];
            return ( 
            <FileWrapper key={fileName} href={currentFile.raw_url} target="__blank">
            <Octicon name="file" ></Octicon>
                <IconTitle> {currentFile.filename} </IconTitle>
            </FileWrapper>
            );
        })}
    </FilesWrapper>
</GistWrapper>;

const GistWrapper = styled.div `
width:100%;
margin-top: 1vw;
border-bottom: 1px solid lightgray;
padding-bottom: 3vw;
`;

const Header = styled.div `
    display: flex;
    justify-content: space-between;
    align-items: center;


`
const Creator = styled.div `
    display:flex;
    align-items: center;

`;
const Avatar = styled.img `
object-fit: contain;
width: 2vw;
height: 2vw;
border-radius: 50%;
margin-right: 0.5vw
`

const Name = styled.div `
 font-size: 1vw;
 color: #2467cf;
`
const IconsWrapper = styled.div `
    width: 50%;
    display: flex;
    justify-content: space-between;
`
const IconWrapper = styled.a `
color: #2467cf;
display: flex;
align-items: center;
font-size: 1vw;
cursor: pointer;
text-decoration: none;

`

const IconTitle = styled.div `
margin-left: 0.25vw;
font-size: 0.75vw;

`

const Dates = styled.div `
    font-size: 0.75vw;
    margin-top: 0.5vw;
`

const Description = styled.div `
    font-size: 1vw;
    margin-top: 1vw;
`
const FilesWrapper = styled.div `
margin-top: 0.75vw;
display: flex;
flex-wrap: wrap;

`
const FileWrapper = styled(IconWrapper) `
    margin-left: 1vw;
`;
export default Gist
