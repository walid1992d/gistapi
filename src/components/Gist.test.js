import { render } from '@testing-library/react';
import React from 'react';
 import Gist from './Gist';
 import {formatDate} from './Gist';
describe("Gist Component", () => {
    const gistData = {
        "url": "https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9",
        "forks_url": "https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9/forks",
        "commits_url": "https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9/commits",
        "id": "f2844d77d44c6203622f8192b20983d9",
        "node_id": "MDQ6R2lzdGYyODQ0ZDc3ZDQ0YzYyMDM2MjJmODE5MmIyMDk4M2Q5",
        "git_pull_url": "https://gist.github.com/f2844d77d44c6203622f8192b20983d9.git",
        "git_push_url": "https://gist.github.com/f2844d77d44c6203622f8192b20983d9.git",
        "html_url": "https://gist.github.com/f2844d77d44c6203622f8192b20983d9",
        "files": {
          "AppStoreLinks.md": {
            "filename": "AppStoreLinks.md",
            "type": "text/markdown",
            "language": "Markdown",
            "raw_url": "https://gist.githubusercontent.com/st3fan/f2844d77d44c6203622f8192b20983d9/raw/fcd0830e5276952da78867a8a43e39d1367a1340/AppStoreLinks.md",
            "size": 211
          }
        },
        "public": true,
        "created_at": "2021-06-03T23:32:41Z",
        "updated_at": "2021-06-03T23:32:41Z",
        "description": "",
        "comments": 0,
        "user": null,
        "comments_url": "https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9/comments",
        "owner": {
          "login": "st3fan",
          "id": 28052,
          "node_id": "MDQ6VXNlcjI4MDUy",
          "avatar_url": "https://avatars.githubusercontent.com/u/28052?v=4",
          "gravatar_id": "",
          "url": "https://api.github.com/users/st3fan",
          "html_url": "https://github.com/st3fan",
          "followers_url": "https://api.github.com/users/st3fan/followers",
          "following_url": "https://api.github.com/users/st3fan/following{/other_user}",
          "gists_url": "https://api.github.com/users/st3fan/gists{/gist_id}",
          "starred_url": "https://api.github.com/users/st3fan/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/st3fan/subscriptions",
          "organizations_url": "https://api.github.com/users/st3fan/orgs",
          "repos_url": "https://api.github.com/users/st3fan/repos",
          "events_url": "https://api.github.com/users/st3fan/events{/privacy}",
          "received_events_url": "https://api.github.com/users/st3fan/received_events",
          "type": "User",
          "site_admin": false
        },
        "truncated": false
      };
    it('should render correctly', () => {
        const {container} = render
          (<Gist gist={gistData}></Gist>)
        expect(container).toMatchSnapshot();
      });

      describe("formatDate()", ()=> {
          it('should format date to mm/dd/yyyy', () => {
            expect(formatDate("2021-06-04T03:10:05.666Z")).toEqual("6/4/2021");
          });
      })
})
