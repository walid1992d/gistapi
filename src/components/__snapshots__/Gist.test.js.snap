// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist Component should render correctly 1`] = `
<div
  className="sc-bdnxRM clTehT"
>
  <div
    className="sc-gtsrHT jABBne"
  >
    <div
      className="sc-dlnjwi dGhCdW"
    >
      <a
        className="sc-gKAaRy jqxyhB"
        href="https://github.com/st3fan"
        target="__blank"
      >
        <img
          className="sc-hKFxyN jmpRVi"
          src="https://avatars.githubusercontent.com/u/28052?v=4"
        />
        <div
          className="sc-eCApnc gcOtLf"
        >
          st3fan
        </div>
      </a>
    </div>
    <div
      className="sc-jSFjdj iARNGw"
    >
      <a
        className="sc-gKAaRy jqxyhB"
      >
        <span
          className="octicon octicon-code"
        />
        <div
          className="sc-iCoGMd lmxDLp"
        >
           
          1
           Files
        </div>
      </a>
      <a
        className="sc-gKAaRy jqxyhB"
        href="https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9/forks"
        target="__blank"
      >
        <span
          className="octicon octicon-repo-forked"
        />
        <div
          className="sc-iCoGMd lmxDLp"
        >
           Forks
        </div>
      </a>
      <a
        className="sc-gKAaRy jqxyhB"
        href="https://api.github.com/gists/f2844d77d44c6203622f8192b20983d9/comments"
        target="__blank"
      >
        <span
          className="octicon octicon-comment"
        />
        <div
          className="sc-iCoGMd lmxDLp"
        >
           Comments
        </div>
      </a>
      <a
        className="sc-gKAaRy jqxyhB"
        href="https://api.github.com/users/st3fan/starred{/owner}{/repo}"
        target="__blank"
      >
        <span
          className="octicon octicon-star"
        />
        <div
          className="sc-iCoGMd lmxDLp"
        >
           Stars
        </div>
      </a>
    </div>
  </div>
  <div
    className="sc-fujyAs fQcKuc"
  >
    Created at: 
    6/4/2021
         Last updated: 
    6/4/2021
  </div>
  <div
    className="sc-pNWdM kNWrll"
  >
    
  </div>
  <div
    className="sc-jrsJWt rKGtP"
  >
    <a
      className="sc-gKAaRy sc-kEqXSa jqxyhB fqSkOs"
      href="https://gist.githubusercontent.com/st3fan/f2844d77d44c6203622f8192b20983d9/raw/fcd0830e5276952da78867a8a43e39d1367a1340/AppStoreLinks.md"
      target="__blank"
    >
      <span
        className="octicon octicon-file"
      />
      <div
        className="sc-iCoGMd lmxDLp"
      >
         
        AppStoreLinks.md
         
      </div>
    </a>
  </div>
</div>
`;
